proposals/ideas/xxx-flashflow.txt

```diff
@@ -0,0 +1,713 @@
+have no list, they check the consensus parameter. If nether exist, then no
+FlashFlow deployment is allowed to measure this relay. Default: empty list.
+
+FFMeasurementPeriod: A relay should expect on average, to be measured by each
+FlashFlow deployment once each measurement period. A relay will not allow
+itself to be measured more than twice by a FlashFlow deployment in any time
+window of this length. Relays should not change this option unless they really
+know what they're doing. Changing it at the relay will not change how often
+FlashFlow will attempt to measure the relay. Possible values are in the range
+[1 hour, 1 month] inclusive. Default: 1 day.
+
+FFBackgroundTrafficPercent: The maximum amount of regular non-measurement
+traffic a relay should handle while being measured, as a percent of total
+traffic (measurement + non-measurement).  This parameter is a trade off
+between having to limit background traffic and limiting how much a relay can
+inflate its result by handling no background traffic but reporting that it has
+done so.  Possible values are in the range [0, 99] inclusive. Default: 25 (a
+maximum inflation factor of 1.33).
+
+FFMaxMeasurementDuration: The maximum amount of time, in seconds, that is
+allowed to pass from the moment the relay is notified that a measurement will
+begin soon and the end of the measurement. If this amount of time passes, the
+relay shall close all measurement connections and exit its measurement mode.
+Note this duration includes handshake time, thus it necessarily is larger than
+the expected actual measurement duration.  Possible values are in the range
+[10, 120] inclusive. Default: 45.
+
+3.1.2 New Cell Types
+
+FlashFlow will introduce a new cell command MEASURE.
+
+The payload of each MEASURE cell consists of:
+
+  Measure command [1 byte]
+  Length          [2 bytes]
+  Data            [Length-3 bytes]
+
+The measure commands are:
+
+  0 -- MSM_PARAMS    [forward]
+  1 -- MSM_PARAMS_OK [backward]
+  2 -- MSM_ECHO      [forward and backward]
+  3 -- MSM_BG        [backward]
+  4 -- MSM_ERR       [forward and backward]
+
+Forward cells are sent from the measurer/coordinator to the relay. Backward
+cells are sent from the relay to the measurer/coordinator.
+
+MSM_PARAMS and MSM_PARAMS_OK are used during the pre-measurement stage to tell the
+target what to expect and for the relay to positively acknowledge the message.
+MSM_ECHO cells are the measurement traffic; the measurer generates them, sends
+them to the target, and the target echos them back. The target send a MSM_BG cells
+once per second to report the amount of background traffic it is handling. MSM_ERR
+cells are used to signal to the other party that there has been some sort of
+problem and that the measurement should be aborted. These cell types are
+described in more detail in the next section.
+
+The only cell type that is sometimes encrypted is MSM_ECHO; all other types
+are never encrypted.
```

Comment:
Aren't the cells sent over TLS connections? It's just that there isn't any additional encryption inside TLS (unlike typical Tor), right?